frontend/src/metabase/hoc/QuestionHandler.jsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router";
+
+import { getQuestion } from "metabase/questions/selectors";
+
+import {
+  fetchQuestion,
+  genUrlBasedQuestion,
+} from "metabase/questions/questions";
+
+const mapStateToProps = state => ({
+  question: getQuestion(state),
+  result: state.questions.currentQuestionResults && state.questions.currentQuestionResults
+});
+
+const getURL = location => location.pathname + location.search + location.hash;
+
+/*
+ * Question handler to handle fetching and URL updates when working with questions for
+ * a particular route
+ */
+class QuestionHandler extends React.Component {
```

Comment:
@tlrobinson (cc: @salsakran if you're interested) I figured one of the easier ways to keep the understanding of "how do I get a question and a result" simpler was to create a component that does most of the work for you. This new thing basically helps guarantee that you'll have a properly instantiated `Question` and a result if you want one and takes care of calling the proper actions. 

Here's an example of it in use in the entity pages branch https://github.com/metabase/metabase/pull/6989/commits/39cf71e52a7ead2c4b5cad683ef8e8630d374f6b

At a high level, you just wrap whatever component you want to use to display a question and its result like so, and you don't need any componentDidMount or any silliness to make sure the things you want will be around.

```js

<QuestionHandler>
{ ({ question, result }) => {
  return(
<div>
{question.card().name}
// ... use a Visualization component etc
<Visualization rawResults={[{
card: question.card(),
data: result.data
}]}
</div>)
}
</QuestionHandler>
```

Needs some hardening, testing etc, but let me know what you think about this approach. The main goal from my perspective was to create a single place where this logic lives that I can then import into anything we're building or prototyping and have high confidence I don't need to dispatch, load or otherwise know anything else and just start working with the data that comes back.